sources/packages/api/src/route-controllers/application/models/application.model.ts

```diff
@@ -103,3 +106,46 @@ export interface ApplicationWithProgramYearDto {
   programYearId: number;
   active: boolean;
 }
+
+export interface SuccessWaitingStatus {
+  success: boolean;
+  waiting: boolean;
+}
+
+export interface ApplicationIncomeVerification {
+  parent1IncomeVerificationStatus: SuccessWaitingStatus;
+  parent2IncomeVerificationStatus: SuccessWaitingStatus;
+  partnerIncomeVerificationStatus: SuccessWaitingStatus;
+  studentIncomeVerificationStatus: SuccessWaitingStatus;
+}
+
+export interface ApplicationSupportingUserDetails {
+  parent1Info: SuccessWaitingStatus;
+  parent2Info: SuccessWaitingStatus;
+  partnerInfo: SuccessWaitingStatus;
+}
+
+export interface InProgressApplicationDetailsAPIOutDTO
+  extends ApplicationIncomeVerification,
+    ApplicationSupportingUserDetails {
+  id: number;
+  applicationStatus: ApplicationStatus;
+  pirStatus: ProgramInfoStatus;
+  PIRDeniedReason: string;
+  offeringStatus: OfferingStatus;
+  exceptionStatus: ApplicationExceptionStatus;
+}
+
+export interface CancelledApplicationDetailsAPIOutDTO {
+  statusUpdatedOn: Date;
+}
+
+export interface ApplicationDetailsAPIOutDTO {
+  applicationNumber: string;
+  applicationInstitutionName: string;
+  applicationStartDate: Date | string;
```

Comment:
I would prefer to have it defined as a `Date`. When needed, convert the string to a Date. The API must ensure that the proper object is returned, independently from specific typed DB column of JSONB.

Reply:
PIR waiting case, dates are taken from the data filed, where it is in string format, 
![image](https://user-images.githubusercontent.com/77353155/192861392-2de4f082-d912-4801-b6f7-2ad10a017d4c.png)

Reply:
did you mean to convert the string to the Date format and send it?